### PR TITLE
fix(defi): deposit percentage precision

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Layers on top of https://github.com/shapeshift/web/pull/3179 bringing more sanity

In the DeFi deposit step, we were doing amount * percentage / 100, which
does exactly this, gives us the result of this operation.

However, making such a division on human amounts will possibly reduce in
more precision than the initial one, which this PR fixes.

Expect more PRs on top of this to:

1. Use the abstracted `handlePercentOptionClick` from
   https://github.com/shapeshift/web/pull/3171 everywhere across the
   DeFi section, to ensure the bug is fixed consistently across deposit
   and withdraws
2. Another PR to bring the crypto/human amount vernacular over the DeFi
   section

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- technically speaking, closes https://github.com/shapeshift/web/issues/3065, but this is not
strictly limited to staking as we use the same implementation
everywhere.

## Risk

None

## Testing

- Crypto and fiat amounts look sane at deposit step when clicking a percent option / max for different assets (test with USDC, ATOM, and some ERC20 of your choice that's not a stablecoin as these tend to have lower precision). Ensure the asset decimal places are honored: 6dp for ATOM and USDC, 18 for most ERC20s

### Engineering

^

### Operations

^
## Screenshots (if applicable)

Develop / Prod

image
image

This diff

image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3180** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3180" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3180?utm_source=stack-comment).

File: src/features/defi/components/Deposit/Deposit.tsx [hunk 1]
```diff
@@ -133,12 +133,18 @@ export const Deposit = ({
 
   const handlePercentClick = useCallback(
     (percent: number) => {
-      const cryptoAmount = bnOrZero(cryptoAmountAvailable).times(percent).precision(asset.precision)
-      const fiatAmount = bnOrZero(cryptoAmount).times(marketData.price)
-      setValue(Field.FiatAmount, fiatAmount.toString(), {
+      // The human crypto amount as a result of amount * percentage / 100, possibly with too many digits
+      const percentageCryptoAmount = bnOrZero(cryptoAmountAvailable).times(percent)
```
Comment (gomesalexandre) [inline src/features/defi/components/Deposit/Deposit.tsx:137]: All renamed for shadowy variable names paranoia - also ditched the precision from this, this was a no-op

File: src/features/defi/components/Deposit/Deposit.tsx [hunk 2]
```diff
@@ -133,12 +133,18 @@ export const Deposit = ({
...
+      const percentageFiatAmount = percentageCryptoAmount.times(marketData.price)
+      const percentageCryptoAmountHuman = percentageCryptoAmount
+        .decimalPlaces(asset.precision)
```
Comment (gomesalexandre) [inline src/features/defi/components/Deposit/Deposit.tsx:140]: Seems like we were using `precision` as a built-in replacement for `decimalPlaces`. `precision` was a no-op in this case

**Review by 0xdef1cafe — APPROVED** [review]
long overdue 😬

**Pull request merged**